### PR TITLE
feat(category): expose a configuration object for `@daffodil/category/state`

### PR DESCRIPTION
**Pull request by damienwebdev**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/graycoreio/daffodil/blob/master/CONTRIBUTING.md#commit
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

```
[ ] Bugfix
[x] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Documentation content changes
[ ] Other... Please describe:
```

## What is the current behavior?
Currently, we force modification of tokens like `DaffDefaultCategoryPageSize`. We should instead be exposing configuration objects of modules.

Fixes: N/A

## What is the new behavior?
We expose a configuration object of the module.

## Does this PR introduce a breaking change?
```
[x] Yes
[ ] No
```

The DaffCategoryRoutingModule has been removed and the `DaffDefaultCategoryPageSize` has moved to `@daffodil/category/state`.

## Other information

### CODE REVIEW

File: libs/category/state/src/category-state.module.ts [hunk 1]
```diff
@@ -16,4 +25,14 @@ import { DAFF_CATEGORY_STORE_FEATURE_KEY } from './reducers/public_api';
...
   ],
 })
-export class DaffCategoryStateModule { }
+export class DaffCategoryStateModule {
+  static forRoot(config: DaffCategoryStateConfiguration = defaultDaffCategoryStateConfiguration): ModuleWithProviders<DaffCategoryStateModule> {
+    return {
+      ngModule: DaffCategoryStateModule,
+      providers: [
+        { provide: DaffCategoryStateConfigurationToken, useValue: config },
+        { provide: DaffDefaultCategoryPageSize, useValue: config.defaultPageSize },
```
Comment (lderrickable) [inline libs/category/state/src/category-state.module.ts:34]: If this is provided here, I don't think `@daffodil/category/routing` should be its own subpackage

File: libs/category/state/src/config/config.ts [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+   * The default size of the number of items to retrieve for a category
+   */
+  defaultPageSize: number;
+}
+
+/**
+ * The default configuration object used to configure the `@daffodil/category/state`
+ * package.
+ */
+export const defaultDaffCategoryStateConfiguration: DaffCategoryStateConfiguration = {
```
Comment (griest024) [inline libs/category/state/src/config/config.ts:23]: Should this be exported as well?

File: libs/category/state/src/category-state.module.ts (hunk 1 above)
Comment (damienwebdev) [reply libs/category/state/src/category-state.module.ts:34]: @griest024 the routing module's were your idea, what do you think?

File: libs/category/state/src/config/config.ts (hunk 2 above)
Comment (damienwebdev) [reply libs/category/state/src/config/config.ts:23]: Yes.

Comment (damienwebdev) [reply libs/category/state/src/config/config.ts:23]: 619b7146

File: libs/category/state/src/category-state.module.ts (hunk 1 above)
Comment (griest024) [reply libs/category/state/src/category-state.module.ts:34]: I don't think it makes sense for guards and resolvers to be in a `state` package. I am still in favor of the `routing` subpackage, despite what the dependency graph looks like.

**Review by griest024 — APPROVED** [review]

File: libs/category/state/src/category-state.module.ts (hunk 1 above)
Comment (lderrickable) [reply libs/category/state/src/category-state.module.ts:34]: so, what I'm saying is that it doesn't make sense to provide the default page size here if we're going to have a routing subpackage. If it doesn't make sense for guards and resolvers to be in a state package, then it also shouldn't make sense to provide the default configuration for the routing subpackage in the state package.

Comment (damienwebdev) [reply libs/category/state/src/category-state.module.ts:34]: I don't think this token is limited solely to the resolvers/guards.

Comment (lderrickable) [reply libs/category/state/src/category-state.module.ts:34]: oh, I hadn't noticed. That makes sense

**Review by lderrickable — APPROVED** [review]

**Pull request merged**